### ISSUE TITLE
New "ntrulike" generator seems broken

### ISSUE THREAD

**Issue by malb** [fplll/fplll#95]
The shortest vector is always (1,1,1,1,1,0,0,0,0,0):

```
In [41]: A = IntegerMatrix.random(5, "ntrulike", bits=10)

In [42]: A
Out[42]: <IntegerMatrix(10, 10) at 0x7fa2fd0b8600>

In [43]: print A
[ 1 0 0 0 0 555 499  68 115 217 ]
[ 0 1 0 0 0 217 555 499  68 115 ]
[ 0 0 1 0 0 115 217 555 499  68 ]
[ 0 0 0 1 0  68 115 217 555 499 ]
[ 0 0 0 0 1 499  68 115 217 555 ]
[ 0 0 0 0 0 727   0   0   0   0 ]
[ 0 0 0 0 0   0 727   0   0   0 ]
[ 0 0 0 0 0   0   0 727   0   0 ]
[ 0 0 0 0 0   0   0   0 727   0 ]
[ 0 0 0 0 0   0   0   0   0 727 ]

In [44]: LLL.reduction(A)
Out[44]: <IntegerMatrix(10, 10) at 0x7fa2fd0b8600>

In [45]: print A
[  -1  -1  -1  -1 -1   0   0   0   0   0 ]
[  17  -4   1 -22  6   2  -4  -7  -2  11 ]
[  -1  22  -6 -17  4   7   2 -11  -2   4 ]
[  22  -6 -17   4 -1   2 -11  -2   4   7 ]
[  -8   4  -3 -19 25  -4   0  -7   5   6 ]
[  -8  12  -1  -4  3   1  14   4   9 -28 ]
[   9  -5 -10  14 -6  -9  21   4  -8  -8 ]
[ -19  -1   1  13  5   2  13 -21   3   3 ]
[ -14   6  -9   5 10   8   8   9 -21  -4 ]
[   9 -11   1 -12 11 126 152 155 148 146 ]
```

**Comment by dstehle** [comment]
That's normal. 
(x^n-1)/(x-1) * h = 0 mod x^n-1, 
because x-1 | h  (due to h(1) = 0).
This is because we forced h(1) to be 0. 
At least, the other vectors, orthogonally to that one, should follow the Gaussian heuristic.

If we force h(1) non-zero, then a very short vector is an (all-a, all-b) vector with a, b
small integers such that h(1) = a/b mod q. It does not impact the expected value of the 
Gaussian heuristic, I think. But it does impact the variance. a,b are much smaller than 
their expected values ~ \sqrt(q) with not so small probability (Gaussian heuristic in dim 2
is much less concentrated than in dimension 2*n). That's the phenomenon we observed
before. 

Both choices are bad, with respect to the Gaussian heuristic. 

What should we do then? Just let h be completely uniform, and warn that the Gaussian 
heuristic doesn't hold for these lattices? Do we have a precise reference explaining this?
